newIDE/app/src/ProjectsStorage/CloudStorageProvider/CloudProjectWriter.js

```diff
@@ -13,13 +13,27 @@ import type { MessageDescriptor } from '../../Utils/i18n/MessageDescriptor.flow'
 import { serializeToJSON } from '../../Utils/Serializer';
 import CloudSaveAsDialog from './CloudSaveAsDialog';
 import { t } from '@lingui/macro';
-import { createZipWithSingleTextFile } from '../../Utils/Zip.js/Utils';
+import {
+  createZipWithSingleTextFile,
+  unzipFirstEntryOfBlob,
+} from '../../Utils/Zip.js/Utils';
 
 const zipProject = async (project: gdProject) => {
   const projectJson = serializeToJSON(project);
   return createZipWithSingleTextFile(projectJson, 'game.json');
 };
 
+const checkZipContent = async (
+  zip: Blob,
+  project: gdProject
+): Promise<boolean> => {
+  const projectJson = serializeToJSON(project);
+  const unzippedProjectJson = await unzipFirstEntryOfBlob(zip);
+  return (
+    unzippedProjectJson === projectJson && !!JSON.parse(unzippedProjectJson)
```

Comment:
JSON.parse will throw an exception if it can parse it. You might want to catch it and return false.